sklearn/feature_selection/variance_threshold.py

```diff
@@ -64,16 +64,15 @@ def fit(self, X, y=None):
         X = check_array(X, ('csr', 'csc'), dtype=np.float64)
 
         if hasattr(X, "toarray"):   # sparse matrix
+            _, self.variances_ = mean_variance_axis(X, axis=0)
             if self.threshold == 0.:
                 mins, maxes = min_max_axis(X, axis=0)
-                self.variances_ = maxes - mins
-            else:
-                _, self.variances_ = mean_variance_axis(X, axis=0)
+                self.variances_ = np.minimum(self.variances_, maxes - mins)
```

Comment:
let's pull this duplicated line out of the if/else and add a comment that we use ptp to avoid numeric precision issues for constant features